### ISSUE TITLE
A transaction's "acquire lock" timeout should be configurable

### ISSUE THREAD

**Issue by alexjpwalker** [graknlabs/client-java#242]
## Problem to Solve

Once https://github.com/graknlabs/grakn/issues/6065 is implemented, ALL transactions will give up attempting to open after 10 seconds if they can't get a lock in time.

The user may want to shorten, or lengthen, this duration, but they can't.

## Current Workaround

Accept the limitations of the system

## Proposed Solution

Add support for the Grakn option (which already exists in `protocol`): `acquire_lock_timeout_millis`.

This change should be propagated to all clients.

**Comment by haikalpribadi** [comment]
Shouldn't we have this issue on the server side? the configuration comes from any client, and takes effect on the server-side, right?

**Comment by alexjpwalker** [comment]
> Shouldn't we have this issue on the server side? the configuration comes from any client, and takes effect on the server-side, right?

My thinking was that it's purely an RPC thing and only affects the clients.

**Comment by haikalpribadi** [comment]
Yeah but I the development still needs to originate from Grakn server actually. So I'll move this to graknlabs/grakn repo, @alexjpwalker.
